ocaml/xcp-rrdd/bin/rrdd/xcp_rrdd.ml

```diff
@@ -545,6 +557,29 @@ let bytes_per_mem_vm = 1024
 
 let mem_vm_writer_pages = ((max_supported_vms * bytes_per_mem_vm) + 4095) / 4096
 
+let read_mem_available () =
+  let meminfo_file = "/proc/meminfo" in
+  let rec read_lines ic =
+    try
+      let line = input_line ic in
+      if String.contains line ':' then
+        let parts = String.split_on_char ':' line in
+        match parts with
+        | key :: value :: _ when String.trim key = "MemAvailable" ->
+            (* Extract the value, remove the 'kB' suffix, trim spaces, and convert to Int64 *)
+            Scanf.sscanf value " %Ld kB" (fun v -> v)
+        | _ ->
+            read_lines ic
+      else
+        read_lines ic
+    with End_of_file -> close_in ic ; Int64.zero
+  in
+  try
+    let ic = open_in meminfo_file in
+    let value = read_lines ic in
```

Comment:
If error raised during read_lines, will the FD be closed?
Suggest to investigate whether `finally` can resolve if any issue.